File: web/test/spec/model/DashboardSpec.js
describe("Dashboard", function() {
  it("should create a dashboard with no monitors", function() {
    var dashboard = new jashboard.model.Dashboard(
      {id: "test_id", name: "test_name", other: "other"}
    );

    expect(dashboard.id).toEqual("test_id");
    expect(dashboard.name).toEqual("test_name");
    expect(dashboard.monitors).toEqual([]);
  });

  it("should add monitor objects", function() {
    var dashboard = new jashboard.model.Dashboard({});
    dashboard.monitors.push({name: "test.monitor.1"});
    dashboard.monitors.push({name: "test.monitor.2"});

    expect(dashboard.monitors.length).toEqual(2);
    expect(dashboard.monitors).toContain({name: "test.monitor.1"});
    expect(dashboard.monitors).toContain({name: "test.monitor.2"});
  });
});

File: server/spec/dashboard_spec.rb
require 'spec_helper'
require 'json_spec'
require 'dashboard_builder'

module Jashboard
  describe Dashboard do
    it("should represent q Dashboard with no monitor ids as json") do
      dashboard = DashboardBuilder.new.
        with_id("test-id").
        with_name("test-name").
        build

      expected_json = %({"id": "test-id", "name": "test-name", "monitor_ids": []})

      dashboard.to_json.should be_json_eql(expected_json).including(:id)
    end
    it("should represent a Dashboard with monitor ids as json") do
      dashboard = DashboardBuilder.new.
        with_id("test-id").
        with_name("test-name").
        with_monitor_id("1").
        with_monitor_id("2").
        build

      expected_json = %({"id": "test-id", "name": "test-name", "monitor_ids": ["1", "2"]})

      dashboard.to_json.should be_json_eql(expected_json).including(:id)
    end
  end
end
